docs/src/prototypes/chatMessageWithPopover/Popover.tsx

```diff
@@ -0,0 +1,183 @@
+import {
+  Menu,
+  Popup,
+  toolbarBehavior,
+  popupFocusTrapBehavior,
+  createComponent,
+  ComponentSlotStyle,
+  ComponentVariablesInput,
+  toolbarButtonBehavior,
+} from '@stardust-ui/react'
+import { ReactChildren } from 'types/utils'
+import * as React from 'react'
+import * as cx from 'classnames'
+
+export interface PopoverProps {
+  className?: string
+}
+
+interface PopoverState {
+  focused: boolean
+  popupOpened: boolean
+}
+
+class Popover extends React.Component<PopoverProps, PopoverState> {
+  state = {
+    focused: false,
+    popupOpened: false,
+  }
+
+  changeFocusState = (isFocused: boolean) => {
+    this.state.focused !== isFocused && this.setState({ focused: isFocused })
+  }
+
+  handleFocus = () => {
+    this.changeFocusState(true)
+  }
+
+  handleBlur = e => {
+    // if e.relatedTarget === null, so the click was outside this container
+    if (!this.state.popupOpened || e.relatedTarget === null) {
+      const shouldPreserveFocusState = e.currentTarget.contains(e.relatedTarget)
+      this.changeFocusState(shouldPreserveFocusState)
+    } else {
+      e.stopPropagation()
+    }
+  }
+
+  handleMenuClick = () => {
+    // close popup when other MenuItem clicked, but the event propagation was stopped
+    this.state.popupOpened && this.setState({ popupOpened: false })
+  }
+
+  menuStyles = ({ theme: { siteVariables } }) => ({
```

Comment:
In my opinion all styles shuold be separate in different file then the definition of the component using them.

Reply:
ok, will move that